pandas/io/parsers.py

```diff
@@ -3420,11 +3428,11 @@ def get_rows(self, n, skiprows=None):
         self.buffer = iter(buffer_rows)
         return detect_rows
 
-    def detect_colspecs(self, n=100, skiprows=None):
+    def detect_colspecs(self, infer_nrows, skiprows=None):
```

Comment:
Is there a reason why you changed this from required to optional? Think that might be causing your test failures

Reply:
It was an oversight. Fixed now. Thanks!